File: course/src/main/java/in/testpress/course/ui/LeaderboardTabAdapter.java
package in.testpress.course.ui;

import android.content.Context;
import android.content.res.Resources;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import in.testpress.core.TestpressSdk;
import in.testpress.course.R;
import in.testpress.models.InstituteSettings;

class LeaderboardTabAdapter extends FragmentPagerAdapter {
    private InstituteSettings instituteSettings;
    private Bundle bundle;

    LeaderboardTabAdapter(Context context, FragmentManager fragmentManager, Bundle bundle) {
        super(fragmentManager);
        this.bundle = bundle;
        instituteSettings = TestpressSdk.getTestpressSession(context).getInstituteSettings();
    }

    @Override
    public int getCount() {
        return 2;
    }

    @Override
    public Fragment getItem(int position) {
        Fragment fragment;
        switch (position) {
            case 0:
                fragment = new RankListFragment();
                break;
            case 1:
                fragment = new TargetThreadFragment();
                break;
            default:
                fragment = new RankListFragment();
                break;
        }
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public CharSequence getPageTitle(final int position) {
        switch (position) {
            case 0:
                return instituteSettings.getLeaderboardLabel();
            case 1:
                return instituteSettings.getThreatsAndTargetsLabel();
            default:
                return null;
        }
    }

}

File: course/src/main/java/in/testpress/course/ui/LeaderboardTabAdapter.kt
package `in`.testpress.course.ui


import `in`.testpress.core.TestpressSdk
import `in`.testpress.models.InstituteSettings
import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class LeaderboardTabAdapter(
    val context: Context,
    fragmentManager: FragmentManager,
    val bundle: Bundle
) : FragmentPagerAdapter(fragmentManager) {

    private var instituteSettings: InstituteSettings =
        TestpressSdk.getTestpressSession(context)?.instituteSettings!!

    override fun getCount(): Int = 2

    override fun getItem(position: Int): Fragment {
        val fragment: Fragment = when (position) {
            0 -> RankListFragment()
            1 -> TargetThreadFragment()
            else -> RankListFragment()
        }
        fragment.arguments = bundle
        return fragment
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            0 -> instituteSettings.leaderboardLabel
            1 -> instituteSettings.threatsAndTargetsLabel
            else -> null
        }
    }
}